### PR TITLE
feat(api): added device get

### PR DESCRIPTION
**Pull request by brianosburn-intel**

### CODE REVIEW

File: src/routes/devices/getAll.ts [hunk 1]
```diff
@@ -32,17 +28,16 @@ export async function getAll (req, res): Promise<void> {
...
+        connectionStatus: req.mpsService.mpsComputerList[m.guid] == null ? 0 : 1,
+        hostname: m.hostname,
+        guid: m.guid,
         metadata: m
       })
     }
 
     if (req.query.status != null) {
-      list = list.filter(x => x.conn === req.query.status)
+      // eslint-disable-next-line eqeqeq
```
Comment (rsdmike) [inline src/routes/devices/getAll.ts:39]: status should be in the validator, and as an number, and this should be === without disabling the rule.

File: src/routes/devices/getAll.ts [hunk 2]
```diff
@@ -32,17 +28,16 @@ export async function getAll (req, res): Promise<void> {
 
     for (const m of metadata) {
       list.push({
-        amtuser: amtCredentials ? amtCredentials[m.guid]?.amtuser : null,
-        conn: req.mpsService.mpsComputerList[m.guid] == null ? 0 : 1,
-        host: m.guid,
-        mpsuser: amtCredentials ? amtCredentials[m.guid]?.mpsuser : null,
-        name: amtCredentials ? amtCredentials[m.guid]?.name : null,
+        connectionStatus: req.mpsService.mpsComputerList[m.guid] == null ? 0 : 1,
```
Comment (rsdmike) [inline src/routes/devices/getAll.ts:31]: nit: can use ??

File: src/models/models.ts [hunk 3]
```diff
@@ -3,11 +3,9 @@
  * SPDX-License-Identifier: Apache-2.0
  **********************************************************************/
 export interface Device {
-  amtuser: string
-  mpsuser: string
-  name: string
-  conn: number
-  host: string
+  connectionStatus: number
```
Comment (rsdmike) [inline src/models/models.ts:6]: much cleaner return type now I think!

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/routes/devices/getAll.ts (hunk 2 above)
Comment (brianosburn-intel) [reply src/routes/devices/getAll.ts:31]: what do you mean?

Comment (rsdmike) [reply src/routes/devices/getAll.ts:31]: actually it the `??` operator won't work here, my mistake

File: src/routes/devices/deviceValidator.ts [hunk 4]
```diff
@@ -10,9 +10,19 @@ export const metadataQueryValidator = (): any => {
     check('tags')
       .optional()
       .isString(),
+    check('status')
+      .optional()
+      .isNumeric,
```
Comment (rsdmike) [inline src/routes/devices/deviceValidator.ts:15]: does this work? I would think this needs to be a function call like the rest? Additionally you should be able to have the validator convert it so you don't need to parseInt later.

Comment (brianosburn-intel) [reply src/routes/devices/deviceValidator.ts:15]: yes it does down.. doesn't work without the parseInt

**Review by rsdmike — APPROVED** [review]

**Pull request merged**